loader_hub/googledrive/base.py

```diff
@@ -0,0 +1,197 @@
+
+    def __init__(self,
+                 credentials_path: str = "credentials.json",
+                 client_secrets_path: str = "client_secrets.json",
+                 token_path: str = "token.json",
+                 file_extractor: Optional[Dict[str,
+                                               Union[str, BaseReader]]] = None,
+                 file_metadata: Optional[Callable[[str], Dict]] = None,) -> None:
+        """Initialize with parameters."""
+        self.credentials_path = credentials_path
+        self.client_secrets_path = client_secrets_path
+        self.token_path = token_path
+        self.file_extractor = file_extractor or DEFAULT_FILE_EXTRACTOR
+        self.file_metadata = file_metadata
+
+    def _get_credentials(self) -> Any:
+        """
+        Reference: https://developers.google.com/drive/api/v3/quickstart/python
+
+        Download the credentials.json file with above reference.
+        Copy credentials.json file and rename it to client_secrets.json file which will be used by pydrive for downloading files.
+
+        So, we need two files:
+        1. credentials.json
+        2. client_secrets.json
+
+        Both 1, 2 are esentially same but needed with two different names according to google-api-python-client, google-auth-httplib2, google-auth-oauthlib and pydrive libraries.
+
+        Returns:
+            credentials, pydrive object
+        """
+        creds = None
+        if os.path.exists(self.token_path):
+            creds = Credentials.from_authorized_user_file(
+                self.token_path, SCOPES)
+        # If there are no (valid) credentials available, let the user log in.
+        if not creds or not creds.valid:
+            if creds and creds.expired and creds.refresh_token:
+                creds.refresh(Request())
+            else:
+                flow = InstalledAppFlow.from_client_secrets_file(
+                    self.credentials_path, SCOPES
+                )
+                creds = flow.run_local_server(port=0)
+            # Save the credentials for the next run
+            with open(self.token_path, "w") as token:
+                token.write(creds.to_json())
+
+        # Authorization for downloading files.
+        # Uses client_secrets.json file for authorization.
+        gauth = GoogleAuth()
+        gauth.LocalWebserverAuth()
+        drive = GoogleDrive(gauth)
+
+        return creds, drive
+
+    def _load_from_file_id(self, file_id: str) -> Document:
+        """Load data from file id
+        Args:
+            file_id: file id of the file in google drive.
+        Returns:
+            Document: Document of text.
+        """
+
+        try:
+
+            # Authenticate and build the Drive API client
+            service = build('drive', 'v3', credentials=creds)
+
+            # Get the file details
+            file = service.files().get(fileId=file_id).execute()
+
+            # Get file name that will be downloaded
+            input_file = file['name']
```

Comment:
It doesn't look like these files are getting cleared out after they're downloaded. Perhaps `tempfile` would be a good option. Check out the example [here](https://github.com/emptycrown/llama-hub/blob/main/loader_hub/s3/base.py) that uses `tempfile.TemporaryDirectory()`. The files are deleted after the code is run.

Reply:
Yeah. This is something that should be taken care of. Will delete files once they are downloaded.